<launch>

	<arg name="num_of_robots" default="7"/>         <!-- Number of robots used all together. -->
    <arg name="num_of_robots1" default="4"/>       <!-- Number of robots used in first group -->
    <arg name="num_of_robots2" default="3"/>        <!-- Number of robots used in second group -->
    <arg name="map_name" default="random_10x10"/>   <!-- Name of the map used. -->
    <arg name="ctrl_loop_freq" default="10"/>       <!-- Frequency used by Reynolds rules. -->
    <arg name="data_stream_freq" default="10"/>     <!-- Position streaming frequency, used by Kalman filter. -->
    <arg name="debug_boids" default="false"/>       <!-- Enable debugging for Reynolds controller node. -->
    <arg name="debug_kalman" default="false"/>      <!-- Enable debugging for Kalman filter node. -->
    <arg name="use_kalman" default="false"/>         <!-- Use either estimated data from Kalman filter or true data from simulator. -->
    <arg name="wait_time" default="0"/>             <!-- During first X seconds of execution, no velocity commands are sent to robots. -->
    <arg name="start_time" default="2"/>  
    <arg name="node_start_delay" default="2.0" />            <!-- During first X seconds after "wait_time", inital velocity commands are sent to robots. -->
    <!-- *********** END OF SECTION *********** -->

    <arg name="map_world" default="$(find sphero_path_following)/resources/$(arg map_name)_$(arg num_of_robots).world"/>
    <arg name="map_yaml" default="$(find sphero_path_following)/maps/$(arg map_name).yaml"/>

    <!-- Set arguments as ros parameter so all nodes can access them. -->
    <param name="num_of_robots" type="int" value="$(arg num_of_robots)"/>
    <param name="num_of_robots1" type="int" value="$(arg num_of_robots1)"/>
    <param name="num_of_robots2" type="int" value="$(arg num_of_robots2)"/>
    <param name="ctrl_loop_freq" type="int" value="$(arg ctrl_loop_freq)"/>
    <param name="data_stream_freq" type="int" value="$(arg data_stream_freq)"/>
    <param name="debug_boids" type="boolean" value="$(arg debug_boids)"/>
    <param name="debug_kalman" type="boolean" value="$(arg debug_kalman)"/>
    <param name="use_kalman" type="boolean" value="$(arg use_kalman)"/>
    <param name="wait_time" type="double" value="$(arg wait_time)"/>
    <param name="start_time" type="double" value="$(arg start_time)"/>
    <param name="run_type" type="string" value="sim"/>



    <node pkg="map_server" type="map_server" args="$(arg map_yaml)" name="map_server"/>
    <node pkg="stage_ros" type="stageros" name="simulator" args="$(arg map_world)"/>

    <node pkg="rqt_gui" type="rqt_gui" name="rqt_gui"/>
    <node pkg="sphero_path_following" type="dynamic_reconfigure_node.py" name="dyn_reconf" output="screen"/>
     <node pkg="sphero_path_following" type="dynamic_reconfigure_node2.py" name="dyn_reconf2" output="screen"/>


    <node pkg="sphero_path_following" type="simulation_tf.py" name="tf_server"/>
    <param name="robot_description" textfile="$(find sphero_path_following)/resources/simple_ball.urdf"/>


    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sphero_path_following)/launch/sphero_sim.rviz"/>


    <node if="$(arg use_kalman)" pkg="sphero_path_following" type="kalman_filter_sim_launch.sh" name="kalman_launcher" args="$(arg num_of_robots)" output="screen"/>
   

</launch>
